### ISSUE TITLE
In Tutorial 1 target sequence len is used at time of evaluation

### ISSUE THREAD

**Issue by dineshkh** [bentrevett/pytorch-seq2seq#122]
Hi,

In Tutorial 1 target sequence is used at time of evaluation.
At the time of evaluation I think we should stop generation of output sequence as soon as we see EOS. But here length of target sequence is used.

**Comment by bentrevett** [comment]
For evaluation (measuring the validation/test loss) we have to always generate the exact same amount of tokens as in the actual target sequence because that is how we measure our loss: cross_entropy(predicted_token, actual_token). 

It does not make sense for us to generate more or less tokens than the ground truth.
